core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -3537,110 +3537,90 @@ class ViewMapping<
 namespace Kokkos {
 namespace Impl {
 
-template <unsigned, class MapType>
-KOKKOS_INLINE_FUNCTION bool view_verify_operator_bounds(const MapType&) {
-  return true;
+template <class Map, class... Indices, std::size_t... Enumerate>
+KOKKOS_FUNCTION bool check_bounds(Map const& map,
+                                  std::index_sequence<Enumerate...>,
+                                  Indices... indices) {
+  return (... && ((std::size_t)indices < map.extent(Enumerate)));
 }
 
-template <unsigned R, class MapType, class iType, class... Args>
-KOKKOS_INLINE_FUNCTION bool view_verify_operator_bounds(const MapType& map,
-                                                        const iType& i,
-                                                        Args... args) {
-  return (size_t(i) < map.extent(R)) &&
-         view_verify_operator_bounds<R + 1>(map, args...);
+template <class... Indices>
+KOKKOS_FUNCTION constexpr char* append_formated_multidimensional_index(
+    char* dest, Indices... indices) {
+  char* d = dest;
+  Kokkos::Impl::strcat(d, "[");
+  (
+      [&] {
+        d += Kokkos::Impl::strlen(d);
+        (void)Kokkos::Impl::to_chars_i(d,
+                                       d + 20,  // 20 digits ought to be enough
```

Comment:
Because it can represent the largest representable signed 64-bit integer.
How would you call that named variable?  I thought the comment was good enough but I am open to change if you have a suggestion that improves readability.

Reply:
I will change it but I am not a big fan of pulling in `<limits>` into our headers (admittedly we already to that from the exec policy header).
Also I think that it does not necessarily convey that at the time that code was written, we picked a number of digits that we think will suffice whereas I thought the code as I initially wrote does.

Reply:
On 2nd thought, not changing unless more reviewers ask for it.

Reply:
```diff
diff --git a/core/src/impl/Kokkos_ViewMapping.hpp b/core/src/impl/Kokkos_ViewMapping.hpp
index 89eff343c..b4894a8ec 100644
--- a/core/src/impl/Kokkos_ViewMapping.hpp
+++ b/core/src/impl/Kokkos_ViewMapping.hpp
@@ -3549,15 +3549,12 @@ KOKKOS_FUNCTION constexpr char* append_formatted_multidimensional_index(
     char* dest, Indices... indices) {
   char* d = dest;
   strcat(d, "[");
-  (
-      [&] {
-        d += strlen(d);
-        to_chars_i(d,
-                   d + 20,  // 20 digits ought to be enough
-                   (size_t)indices);
-        strcat(d, ",");
-      }(),
-      ...);
+  for (auto index : {static_cast<size_t>(indices)...}) {
+    d += strlen(d);
+    constexpr int max_digits = 20;  // enough for uint64_t
+    to_chars_i(d, d + max_digits, index);
+    strcat(d, ",");
+  }
   d[strlen(d) - 1] = ']';  // overwrite trailing comma
   return dest;
 }
```
would also look better to me.

Reply:
or just the `max_digits` part.

Reply:
The `static_cast` would potentially lead to poor quality error messages (see https://github.com/kokkos/kokkos/pull/6861#discussion_r1517992352).
I don't think the `max_digits` is better than the current version and I am not planning to change unless more reviewers ask me to change it.